### PR TITLE
Add several spelling corrections

### PR DESCRIPTION
**Pull request by luzpaz**

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -18893,6 +18898,8 @@ itention->intention
 itentional->intentional
 itentionally->intentionally
 itentionaly->intentionally
+iteraction->interaction
+iteractions->interactions
 iteratable->iterable
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:18903]: If you do not mind suggesting `iteration` as well?
```text
iteraction->interaction, iteration,
iteractions->interactions, iterations,
```

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -24770,6 +24777,7 @@ probabilty->probability
 probabily->probability, probably,
 probablay->probably
 probablistic->probabilistic
+probablity->probability
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:24780]:
```text
probablities->probabilities
```
https://github.com/search?q=probablities&type=code (22K hits)

**Review by vikivivi — CHANGES_REQUESTED** [review]

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (luzpaz) [reply codespell_lib/data/dictionary.txt:18903]: [98365e9](https://github.com/codespell-project/codespell/pull/2423/commits/98365e91351c7cc1035d18c6eb7c849f2bf0b91b)

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (luzpaz) [reply codespell_lib/data/dictionary.txt:24781]: [98365e9](https://github.com/codespell-project/codespell/pull/2423/commits/98365e91351c7cc1035d18c6eb7c849f2bf0b91b)

**Pull request merged**